base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -140,31 +146,22 @@ pub trait BlockchainBackend: Send + Sync {
     /// Checks to see whether the given key exists in the back end. This function should only fail if there is an
     /// access or integrity issue with the back end.
     fn contains(&self, key: &DbKey) -> Result<bool, ChainStorageError>;
-    /// Fetches the merklish root for the MMR tree identified by the key. This function should only fail if there is an
-    /// access or integrity issue with the back end.
-    fn fetch_mmr_root(&self, tree: MmrTree) -> Result<HashOutput, ChainStorageError>;
-    /// Returns only the MMR merkle root without the state of the roaring bitmap.
-    fn fetch_mmr_only_root(&self, tree: MmrTree) -> Result<HashOutput, ChainStorageError>;
-    /// Fetches the merklish root for the MMR tree identified by the key after the current additions and deletions have
-    /// temporarily been applied. Deletions of hashes from the MMR can only be applied for UTXOs.
-    fn calculate_mmr_root(
-        &self,
-        tree: MmrTree,
-        additions: Vec<HashOutput>,
-        deletions: Vec<HashOutput>,
-    ) -> Result<HashOutput, ChainStorageError>;
-    /// Constructs a merkle proof for the specified merkle mountain range and the given leaf position.
-    fn fetch_mmr_proof(&self, tree: MmrTree, pos: usize) -> Result<MerkleProof, ChainStorageError>;
-    /// Fetches the `MerkleCheckPoint` corresponding to the given height. In pruned mode, the underlying database may
-    /// not be able to provide the checkpoint as it has been merged into the base checkpoint. In this case a
-    /// `BeyondPruningHorizon` error is returned.
-    fn fetch_checkpoint_at_height(&self, tree: MmrTree, height: u64) -> Result<MerkleCheckPoint, ChainStorageError>;
-    /// Fetches the `MerkleCheckPoint` at the given index
-    fn fetch_checkpoint_at_index(
+
+    fn fetch_header_accumulated_data(
```

Comment:
I planned on putting the calculated achieved difficulty in `header_accumulated_data` as well. I thought it could be retrieved once per header before calling the validator stack. I don't think it's leaking much. Right now it's effort to change it to just retrieve the `total_kernel_commitment_sum` so not really saving any future effort by changing it now